timbuctoo-instancev4/src/main/java/nl/knaw/huygens/timbuctoo/bulkupload/loaders/csv/CsvLoader.java

```diff
@@ -0,0 +1,89 @@
+package nl.knaw.huygens.timbuctoo.bulkupload.loaders.csv;
+
+import nl.knaw.huygens.timbuctoo.bulkupload.InvalidFileException;
+import nl.knaw.huygens.timbuctoo.bulkupload.loaders.Loader;
+import nl.knaw.huygens.timbuctoo.bulkupload.parsingstatemachine.Importer;
+import nl.knaw.huygens.timbuctoo.util.Tuple;
+import org.apache.commons.csv.CSVFormat;
+import org.apache.commons.csv.CSVParser;
+import org.apache.commons.csv.QuoteMode;
+
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * A CsvLoader parses CSV files and treats each file as a separate collection.
+ * The default format is Excel's CSV format, with a header row.
+ */
+public class CsvLoader implements Loader {
+  private final CSVFormat format;
+
+  public CsvLoader(Map<String, String> config) {
+    CSVFormat format = CSVFormat.EXCEL;
+    if (config.containsKey("delimiter")) {
+      format = format.withDelimiter(config.get("delimiter").charAt(0));
```

Comment:
yes, that would be safer in all cases... I had that typed in but reverted it due to a temporary lack of pro-activity on my part.